### PR TITLE
More tests and proptests

### PR DESCRIPTION
**Pull request by edg-l**
Adds tests and proptests for the binary operations

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe):

### CODE REVIEW

**Comment by edg-l** [comment]
This reworks the way type id and names are stored (imho easier now), so that it works better with the debug code.

I also changed the return type from always being a struct to being only a struct if the function returns more than 1 value.

I need to fix a couple bugs (2 tests don't pass) and will be ready

File: core/tests/proptests.rs [hunk 1]
```diff
@@ -46,13 +49,86 @@ fn simple_addition() {
...
+    let return_value = BigInt::from_str_radix(output, 16).unwrap();
+
+    let prime = get_prime();
+
+    let lhs = BigInt::from_str_radix(lhs, 10).unwrap();
+    let rhs = BigInt::from_str_radix(rhs, 10).unwrap();
+    let mut expected = bigint_op(lhs, rhs) % prime;
+    let zero = BigInt::zero();
+    let two = BigInt::from(2).pow(return_value.bits() as u32);
+    expected += if expected < zero { two } else { zero };
```
Comment (LucasLvy) [inline core/tests/proptests.rs:62]:
```suggestion
    let two = BigInt::from(2).pow(x.bits() as u32);
    expected = expected.modpow(&BigInt::from(1), &two);
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]
How about having only 1 template where the operation is also set in the context ?

**Review by LucasLvy — APPROVED** [review]
lgtm

**Pull request merged**